carbon_black_cloud/icon_carbon_black_cloud/connection/connection.py

```diff
@@ -0,0 +1,79 @@
+
+        try:
+            result.raise_for_status()
+        except Exception as e:
+            if result.status_code == 400:
+                raise PluginException(cause="400 Bad Request",
+                                      assistance="Verify that your request adheres to API documentation.",
+                                      data=result.text)
+            if result.status_code == 401:
+                raise PluginException(cause="Authentication Error",
+                                      assistance="Please verify that your secret_key and api_id values are correct.",
+                                      data=result.text)
+            if result.status_code == 403:
+                raise PluginException(cause="The specified object cannot be accessed or changed.",
+                                      assistance="If it has a Custom access level, check it has been assigned the correct RBAC permissions. If it is an API, SIEM or LIVE_RESPONSE type key, verify it is the right key type for the API in use.",
+                                      data=result.text)
+            if result.status_code == 404:
+                raise PluginException(cause="The object referenced in the request cannot be found.",
+                                      assistance="Verify that your request contains objects that haven’t been deleted. Verify that the org_key in the URL is correct.",
+                                      data=result.text)
+            if result.status_code == 409:
+                raise PluginException(cause="Either the name you chose already exists, or there is an unacceptable character used.",
+                                      assistance="Change any spaces in the name to underscores. Look through your list of API Keys and see if there is an existing key with the same name.",
+                                      data=result.text)
+            if result.status_code == 503:
+                raise PluginException(cause="Cannot return object at this moment because service is unavailable.",
+                                      assistance="This can happen if too many file downloads are happening at the same time. You can try later.",
+                                      data=result.text)
+            self.logger.error(f"Exception was:\n{e}")
+            self.logger.error(f"Result text was:\n{result.text}")
+            raise PluginException(PluginException.Preset.UNKNOWN)
+
+        return result.json()
+
+
+    def test(self):
+        device_endpoint = "/device"
+        endpoint = self.base_url + device_endpoint
+
+        self.logger.info(endpoint)
+        result = requests.get(endpoint, headers=self.headers)
+        try:
+            result.raise_for_status()
+        except Exception as e:
+            raise ConnectionTestException(cause="Connection test to Carbon Black Cloud failed.\n",
+                                          assistance=f"{result.text}\n",
```

Comment:
Does result.text tell the user enough information? like Credentials are invalid, please enter and try again kinda thing

Reply:
It seems to be pretty clear on authentication errors.